props/props.md

```diff
@@ -0,0 +1,619 @@
+    )
+}
+```
+<!-- prettier-ignore-end -->
+
+The function call `studentData.map(...)` sets `studentComponents` to an array of JSX objects. It does this by looking at each object in `studentData`, referring to each object as `student` one at a time.
+
+Each `student` object has the keys `nameData` and `emailData`. We read these values with `student.nameData` and `student.emailData`. We embed those values into a new JSX object, which is a `Student` component with the properties `name` and `email`.
+
+Our resulting webapp looks identical to before! However, now Sofia's `StudentList` component is more robust, and can easily render all students, even if `studentData` grows.
+
+## More Practice: Passing `studentData` From `App` to `StudentList`
+
+Let's practice using `props` one more time!
+
+Right now, `studentData` is defined and used in the `StudentList` component.
+
+Sofia anticipates that, actually, her `App` component should hold and manage the `studentData`... but `StudentList` should continue to use it.
+
+She can move `studentData` into `App`, and pass it to `StudentList` through `props`!
+
+### Sending `props` From `App`
+
+The `App` component can define `studentData`, and then send it as a property.
+
+<!-- prettier-ignore-start -->
+```js
+function App() {
+  const studentData = [
+    {
+      nameData: 'Ada',
+      emailData: 'ada@dev.org'
+    },
+    {
+      nameData: 'Soo-ah',
+      emailData: 'sooah@dev.org'
+    },
+    {
+      nameData: 'Chrissy',
+      emailData: 'chrissy@dev.org'
+    }
+  ]
+
+  return (
+    <main>
+      <h1>Attendance</h1>
+      <ClassInfo></ClassInfo>
+      <StudentList students={studentData}></StudentList>
+    </main>
+  );
+}
+```
+<!-- prettier-ignore-end -->
+
+In this situation, `StudentList` is receiving a prop named `students`.
+
+### Reading `props` in `StudentList`
+
+Like before, in order for `StudentList` to read props, Sofia must:
+
+- Add `props` as an expected parameter for the `StudentList` component function
+- Modify her code to use `props.students`
+
+<!-- prettier-ignore-start -->
+```js
+const StudentList = (props) => {
+    const studentComponents = props.students.map(student => {
+        return <Student name={student.nameData} email={student.emailData}></Student>
+    });
+
+    return (
+        <section>
+            <h2>Student List</h2>
+            <ul>
+                {studentComponents}
+            </ul>
+        </section>
+    )
+}
+```
+<!-- prettier-ignore-end -->
+
+Even after this refactor, Sofia's webapp looks the same as before. Great!
+
+### !callout-warning
+
+## Addressing "Warning: Each child in a list should have a unique "key" prop."
+
+If we check our console, we may see the warning "Warning: Each child in a list should have a unique "key" prop."
+
+![An attendance web app with the browser dev tools open. The console of the dev tools reads "Warning: Each child in a list should have a unique "key" prop."](../assets/props_props_warning-key.png)  
```

Comment:
technically, this screen shot doesn't fully match the error that would appear if Student is wrapped in an li, but it might not noticeable enough to warrant updating this image.